application/src/main/java/uk/gov/hmcts/dm/service/BlobStorageReadService.java

```diff
@@ -16,26 +24,117 @@
 @Transactional
 public class BlobStorageReadService {
 
+
     private final BlobContainerClient cloudBlobContainer;
+    private HttpServletRequest request;
+    private static final int DEFAULT_BUFFER_SIZE = 20480; // ..bytes = 20KB.
 
     @Autowired
-    public BlobStorageReadService(BlobContainerClient cloudBlobContainer) {
+    public BlobStorageReadService(BlobContainerClient cloudBlobContainer, HttpServletRequest request) {
         this.cloudBlobContainer = cloudBlobContainer;
+        this.request = request;
     }
 
     public void loadBlob(DocumentContentVersion documentContentVersion, OutputStream outputStream) {
         log.debug("Reading document version {} from Azure Blob Storage...", documentContentVersion.getId());
-        BlockBlobClient blob = loadBlob(documentContentVersion.getId().toString());
-        blob.download(outputStream);
+        BlockBlobClient blobClient = blockBlobClient(documentContentVersion.getId().toString());
+        blobClient.download(outputStream);
         log.debug("Reading document version {} from Azure Blob Storage: OK", documentContentVersion.getId());
     }
 
-    private BlockBlobClient loadBlob(String id) {
+    /**
+     * Streams the content of a document from blob storage. Processes a Range request when possible
+     */
+    public void loadBlob(DocumentContentVersion documentContentVersion, HttpServletResponse response) throws IOException {
```

Comment:
I would maybe restructure this method so that it basically just becomes:

``` return request.getHeader(HttpHeaders.RANGE) == null ? loadFullBlob(....) : loadPartialBlob(...);```

as it stands you get the blobClient and then don't use it if there is no range specified. Best to move all that into separate methods.

Reply:
I see that `processRangeRequest` also calls `loadBlob` in some cases  - perhaps you should use this method to get the client, then pass that client to `loadFullBlob` or `loadPartialBlob`

Reply:
sounds good, I'll take a look 👍